src/lib/datepicker/datepicker.ts

```diff
@@ -188,18 +203,29 @@ export class MdDatepicker implements OnDestroy {
 
   /** Open the calendar as a dialog. */
   private _openAsDialog(): void {
-    this._dialogRef = this._dialog.open(this.calendarTemplate);
+    let config = new MdDialogConfig();
+    config.viewContainerRef = this._viewContainerRef;
+    config.disableFocusTrap = true; // We'll handle focus trapping ourselves.
```

Comment:
Can we make `disableFocusTrap` internal? I don't want to encourage people to do this and mess up the a11y